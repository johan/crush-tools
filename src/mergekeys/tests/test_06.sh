test_number=06
description="case-insensitive string collation (assumes local en_US.iso88591)"

# only run locale-sensitive tests when they can be expected to work
if [ "`which locale 2>/dev/null`" ] &&
   [ "`locale -a | grep en_US.iso88591`" ]; then
  export LC_ALL="en_US.iso88591"
  export LC_COLLATE="en_US.iso88591"
  locale_works=1
fi

if [ ! $locale_works ]; then
  test_status $test_number all "$description" SKIP
else
  for i in `seq 0 $((${#test_variants[*]} - 1))`; do
    outfile="$test_dir/test_$test_number.${test_variants[$i]}.actual"
    expected=$test_dir/test_$test_number.${test_variants[$i]}.expected
    $bin ${test_variants[$i]} \
         -o "$outfile" \
         "$test_dir/test_$test_number.a" \
         "$test_dir/test_$test_number.b"

    if [ $? -ne 0 ] ||
       [ "`diff -q $outfile $expected`" ]; then
      test_status $test_number $i "$description (${variant_desc[$i]})" FAIL
    else
      test_status $test_number $i "$description (${variant_desc[$i]})" PASS
      rm "$outfile"
    fi
  done
fi
