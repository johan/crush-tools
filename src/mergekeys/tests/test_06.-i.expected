Month	Site	Ad Size	Ad Type	Country Category	Impressions	Clicks
June	Page 3	468x60	Standard	US	1008405	9
June	The	125x125	Exclusive	EU	750	6
June	The	125x125	Exclusive	Others	1042	25
